tier2/steam_api.cpp

```diff
@@ -100,6 +106,50 @@ void LoadSteamClient017()
 	}
 
 	steamClientModule = Sys_LoadModule(libpath.c_str());
+#elif defined(OSX)
+	std::string target_process("steam_osx");
+	
+	// To get a list of mapped libraries in any given executable we need to first access its
+	// task port with task_for_pid-allow, this requires a special entitlement and code signing
+	// which is overkill for this narrow purpose. The quick and easy solution to get around
+	// that and find this library dynamically regardless of installation path is to find an
+	// instance of steam_osx, strip the name, and substitute steamclient.dylib in its place.
+	// This method will always work because steam_osx is required for any game to function and
+	// they're both contained within the same application bundle. Sys_LoadModule will always be
+	// passed the correct path to steamclient.dylib or somehow even in the event of a fail condition,
+	// the path to an unknown process is given, failing silently.
+	int count = proc_listpids(PROC_ALL_PIDS, 0, nullptr, 0) / sizeof(pid_t);
```

Comment:
I think `proc_listpids` returns number of processes so there is no need to divide by `sizeof(pid_t)`.
https://gist.github.com/nguyen-phillip/de66b0ea2144e20ddd844c41c9d93eb9#file-proc-c-L50

Reply:
[https://opensource.apple.com/source/xnu/xnu-4903.221.2/bsd/kern/proc_info.c.auto.html](url). This is the internal part that gets called when we use the library version. When an empty buffer is passed into `proc_listpids` to query the number of processes:
```
	/* if the buffer is null, return num of procs */
	if (buffer == (user_addr_t)0) {
		*retval = ((nprocs + 20) * sizeof(int));
		return(0);
	}
```
We get back the product of the total number of processes and the effective size of `pid_t`.
[https://opensource.apple.com/source/lldb/lldb-76/tools/darwin-threads/examine-threads.c.auto.html](url). It's also how Apple themselves handle counting the number of processes.

Reply:
I had a bit of a worry myself when double and triple checking the retval of `proc_listpids` when accounting for every way it can fail before pushing a commit.